torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -400,8 +428,31 @@ void ProcessGroupAgent::handleSend(const SendWork& work) {
     pendingSends.emplace_back(pg_->send(preamble, dst, dst /* channelTag */));
     pendingSends.emplace_back(pg_->send(payload, dst, dst /* channelTag */));
   }
+  // Write pendingSends to a global map so that they can be interrupted by
+  // ::shutdown().
+  std::unique_lock<std::mutex> pendingSendGuard(pendingSendMutex_);
```

Comment:
this is fine, though might be slightly easier to follow the locking if using blocks:

```
   {
      std::unique_lock<std::mutex> pendingSendGuard(pendingSendMutex_);
      for (auto& p : pendingSends) {
        currentPendingSends_[dst].insert(p);
      }
    }
```